src/cmd/run.js

```diff
@@ -117,6 +121,18 @@ export default async function run(
   const customPrefs = pref;
   const manifestData = await getValidatedManifest(sourceDir);
 
+  const profileDir = firefoxProfile || chromiumProfile;
+
+  if (profileCreateIfMissing && profileDir) {
```

Comment:
while I was looking this part one more time I did notice another corner case that may be worth to consider (and eventually handle in a follow up):

if the user does call web-ext run with --profile-create-if-missing but without any profile parameter we currently just ignore the --profile-create-if-missing, it isn't strictly wrong and so it is not really a bit deal, but another option would be to throw an explicit `UsageError` to make the user aware that --profile-create-if-missing should be paired with --firefox-profile or --chromium-profile).

what do you think?

as I did mention it is not a really big deal and so it seems ok to defer it to a separate follow up.

Reply:
Sure. That does make sense. I can do that in a follow up PR. Should we be creating a separate issue for that? Or should I just send another PR after this one is merged?

Reply:
> Sure. That does make sense. I can do that in a follow up PR. Should we be creating a separate issue for that? Or should I just send another PR after this one is merged?

@akhilpanchal thanks! just a separate PR for that part would be fine

in this one let's just cover the nits on the test, and after that I'll just merge it and you can open the follow up.

Reply:
@rpl 
I have addressed all the nits. Please let me know if I missed something.
Once this is merged, I will create a follow up for this enhancement.
Also, you mentioned making changes to the [mozilla/extension-workshop](https://github.com/mozilla/extension-workshop) repo. Should that be also linked to an issue in that repo?